tools/analysis/analysis_test.go

```diff
@@ -259,6 +259,168 @@ func TestCheckError(t *testing.T) {
 	require.ErrorAs(t, err, &checkerError)
 }
 
+func TestHandledParserError(t *testing.T) {
+
+	t.Parallel()
+
+	contractAddress := common.MustBytesToAddress([]byte{0x1})
+	contractLocation := common.AddressLocation{
+		Address: contractAddress,
+		Name:    "ContractA",
+	}
+	const contractCode = `
+	  access(all) contract ContractA {
+	    init() {
+	      ???
+	    }
+	  }
+	`
+
+	config := &analysis.Config{
+		Mode: analysis.NeedSyntax,
+		ResolveCode: func(
+			location common.Location,
+			importingLocation common.Location,
+			importRange ast.Range,
+		) ([]byte, error) {
+			switch location {
+			case contractLocation:
+				return []byte(contractCode), nil
+
+			default:
+				require.FailNow(t,
+					"import of unknown location: %s",
+					"location: %s",
```

Comment:
`FailNow` doesn't format, so the message (second argument) is used as-is, the `%s` format specifier isn't used.

Use `FailNowf` instead

```suggestion
				require.FailNowf(t,
					"import of unknown location: %s",
```

Reply:
ok sounds good, I had just copied this from other tests so I have no switched them all over.  It looks like `FailNow`/`FailNowf` require both a failure message and a formatted message, so I've switched it to this.

`func require.FailNowf(t require.TestingT, failureMessage string, msg string, args ...interface{})`